### PR TITLE
usb: Add API for getting status of remote wakeup

### PR DESCRIPTION
**Pull request by eobalski**
This change adds a posibility to check the status of
remote wakeup feature.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

### CODE REVIEW

**Review by MarekPieta — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Review by rakons — APPROVED** [review]

File: include/usb/usb_device.h [hunk 1]
```diff
@@ -433,6 +433,18 @@ bool usb_transfer_is_busy(uint8_t ep);
...
+ * @brief Get status of the USB remote wakeup feature
+ *
+ * This feature must be enabled in configuration, otherwise
+ * it will always return -ENOTSUP error.
+ *
+ * @return 0 if remote wakeup is supported and enabled by the host,
+ *         -ENOTSUP if remote wakeup is not supported,
+ *         -EACCES if remote wakeup is supported, but not enabled.
+ */
+int usb_get_remote_wakeup_status(void);
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply include/usb/usb_device.h:446]: With https://github.com/zephyrproject-rtos/zephyr/pull/40047, I suggest to simplify it to just return a bool when feature is set by the host and false in all other cases. Whether the device controller is capable or not can be checked in an application with IS_ENABLED(CONFIG_USB_DEVICE_REMOTE_WAKEUP).

[the remark this answers is not in the document]
Comment (emob-nordic) [reply include/usb/usb_device.h:446]: I will rewrite this PR, once #40047 is in.

**Comment by carlescufi** [comment]
@emob-nordic please rebase now that #40047 is in

**Comment by emob-nordic** [comment]
Simplified as much as possible. @jfischer-no @anangl please have a look.

File: include/usb/usb_device.h [hunk 2]
```diff
@@ -433,6 +433,15 @@ bool usb_transfer_is_busy(uint8_t ep);
  */
 int usb_wakeup_request(void);
 
+/**
+ * @brief Get status of the USB remote wakeup feature
+ *
+ * Function to get status of remote wakeup feature.
```
Comment (anangl) [inline include/usb/usb_device.h:439]: Is this description actually needed? It does not bring much more than the brief one.

Comment (emob-nordic) [reply include/usb/usb_device.h:439]: At first it was more useful, but the PR has changes a lot since last change in upstream. I will remove this.

Comment (emob-nordic) [reply include/usb/usb_device.h]: Done

File: include/usb/usb_device.h [hunk 3]
```diff
@@ -433,6 +433,13 @@ bool usb_transfer_is_busy(uint8_t ep);
  */
 int usb_wakeup_request(void);
 
+/**
+ * @brief Get status of the USB remote wakeup feature
+ *
+ * @return true if remote wakeup enabled by the host, false otherwise.
```
Comment (anangl) [inline include/usb/usb_device.h:439]:
```suggestion
 * @return true if remote wakeup has been enabled by the host, false otherwise.
```

**Review by anangl — APPROVED** [review]

**Comment by emob-nordic** [comment]
@jfischer-no Could you have a look, please?

**Review by jfischer-no — APPROVED** [review]

**Comment by emob-nordic** [comment]
@carlescufi can we merge already? There is nothing that could possibly fail in CI.

**Pull request merged**